compiler/compile/OMRCompilation.cpp

```diff
@@ -1855,14 +1856,16 @@ void OMR::Compilation::switchCodeCache(TR::CodeCache *newCodeCache)
    _numReservedIPICTrampolines = 0;
    if ( self()->cg()->committedToCodeCache() || !newCodeCache )
       {
-      if (self()->getErrorCode() == COMPILATION_SUCCEEDED)
+      traceMsg(self(), "Already committed to current code cache");
+
+      if (newCodeCache)
          {
-         if (newCodeCache)
-            self()->setErrorCode(COMPILATION_ILLEGAL_CODE_CACHE_SWITCH);
-         else
-            self()->setErrorCode(COMPILATION_NULL_SUBSTITUTE_CODE_CACHE);
+         self()->setErrorCode(COMPILATION_ILLEGAL_CODE_CACHE_SWITCH);
+         throw TR::RecoverableCodeCacheError();
          }
-      self()->fe()->outOfMemory(self(), "Already committed to current code cache");
+
+      self()->setErrorCode(COMPILATION_NULL_SUBSTITUTE_CODE_CACHE);
+      throw TR::CodeCacheError();
```

Comment:
In addition, the justification comes from the fact that if we're switching to a NULL cache, we likely couldn't allocate a new one and we're likely to have exhausted our code area.